src/GraphQL/Types/TypeInformation.cs

```diff
@@ -0,0 +1,259 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace GraphQL.Types
+{
+    /// <summary>
+    /// Parses a <see cref="System.Type"/> along with nullability information into its constituent parts
+    /// in preparation for creating a graph type that represents such a type.
+    /// </summary>
+    public class TypeInformation
+    {
+        /// <summary>
+        /// The member being inspected.
+        /// </summary>
+        public MemberInfo MemberInfo { get; }
+
+        /// <summary>
+        /// Indicates that this is an input type (an argument); false for output types.
+        /// </summary>
+        public bool IsInputType { get; }
+
+        /// <summary>
+        /// The underlying CLR type represented. This might be the underlying type of a <see cref="Nullable{T}"/>
+        /// or the underlying type of a <see cref="IEnumerable{T}"/>.
+        /// </summary>
+        public Type Type { get; set; }
+
+        /// <summary>
+        /// Indicates if the underlying type is nullable.
+        /// </summary>
+        public bool IsNullable { get; set; }
+
+        /// <summary>
+        /// Indicates that this represents a list of elements.
+        /// </summary>
+        public bool IsList { get; set; }
```

Comment:
IsList / ListIsNullable looks some strange to me but I nearly understand what happens here. I do not want to delay this PR more time. I will see later how this code works under the debugger with concrete values.